app/models/moderation.rb

```diff
@@ -77,4 +78,12 @@ def send_message_to_moderated
 
     m.save
   end
+
+protected
+
+  def one_foreign_key_present
+    fields = [comment, domain, story, tag, user]
+    fk_count = fields.inject(0) {|result, fk| fk.nil? ? result : result + 1 }
```

Comment:
Using the model attribute like this instead of the `_id` column will cause Rails to query for and instantiate the related object(s) unnecessarily. 

You can also express this more directly with `.compact.length`.

Reply:
> Using the model attribute like this instead of the _id column will cause Rails to query for and instantiate the related object(s) unnecessarily.

I changed everything to _id.

> You can also express this more directly with .compact.length.

I didn't know of this, thanks.